snaps_boot/provision/rebar_utils.py

```diff
@@ -444,8 +444,31 @@ def __create_machine_params(boot_conf, public_key):
 
     http_proxy = prov_conf['PROXY']['http_proxy']
     https_proxy = prov_conf['PROXY']['https_proxy']
-    out.append(ParamsModel(name='post/http-proxy', value=http_proxy))
-    out.append(ParamsModel(name='post/https-proxy', value=https_proxy))
+    apt_proxy = prov_conf['PROXY']['ngcacher_proxy']
+    if http_proxy:
+        out.append(ParamsModel(name='post/http-proxy', value=http_proxy))
+    if https_proxy:
+        out.append(ParamsModel(name='post/https-proxy', value=https_proxy))
+    if apt_proxy:
+        out.append(ParamsModel(name='post/ngcacher-proxy', value=apt_proxy))
+
+    mellanox_install = False
+    # 'POST' section is optional
+    if prov_conf.get('POST'):
+        post_confs = prov_conf['POST']
+        # 'mellanox' section is optional
+        if post_confs.get('mellanox'):
```

Comment:
See comment in hosts.yaml. If going more generic is a viable option, this is where you would implement.